### PR TITLE
JW-103: vif utility port

### PR DESCRIPTION
**Pull request by sodar**
Before review PR must be rebased after:

* ~~merging https://github.com/codilime/contrail-vrouter/pull/70~~
* ~~merging https://github.com/codilime/contrail-vrouter/pull/59~~

### CODE REVIEW

**Comment by m-kostrzewa** [comment]
Janie, przetestuj proszę

File: utils/vif.c [hunk 1]
```diff
@@ -1379,8 +1417,67 @@ rate_stats(struct nl_client *cl, unsigned int vr_op)
...
+    char fmt[80] = { 0 };
+    int ret = 0;
+
+    print_number_interface = MAX_OUTPUT_IF;
+    ignore_number_interface = 0;
+    first_rate_iter = true;
+
+    /* Wait for 1s so rate_stats_diff could get non-zero diff_ms */
+    last_timer_value = GetTickCount64();
+    Sleep(1000);
```
Comment (Attil) [inline utils/vif.c]: You can just call `Sleep` in every iteration instead of this.

File: utils/vif.c [hunk 2]
```diff
@@ -1379,8 +1417,67 @@ rate_stats(struct nl_client *cl, unsigned int vr_op)
...
+
+    while (true) {
+        if (!first_rate_iter) {
+            Sleep(1000);
+        }
+
+        /* Size of the terminal does not limit us on Windows. */
```
Comment (Attil) [inline utils/vif.c]: Why?

File: utils/vif.c [hunk 3]
```diff
@@ -1379,8 +1417,67 @@ rate_stats(struct nl_client *cl, unsigned int vr_op)
...
+    print_number_interface = MAX_OUTPUT_IF;
+    ignore_number_interface = 0;
+    first_rate_iter = true;
+
+    /* Wait for 1s so rate_stats_diff could get non-zero diff_ms */
+    last_timer_value = GetTickCount64();
+    Sleep(1000);
+
+    while (true) {
+        if (!first_rate_iter) {
```
Comment (Attil) [inline utils/vif.c]: `first_rate_iter` variable is not needed I think. At least not here.

Comment (sodar) [inline utils/vif.c]: `first_rate_iter` is used in other functions to check if `vif` can calculate a difference between last and current interface stats.

File: utils/vif.c (hunk 2 above)
Comment (sodar) [inline utils/vif.c]: On Linux `print_number_interfaces` is based on height of the terminal and `vif` uses keyboard input to scroll the list of vifs. On Windows we do not implement the same mechanic, thus we output maximum amount.

File: utils/vif.c [hunk 4]
```diff
@@ -1379,8 +1417,62 @@ rate_stats(struct nl_client *cl, unsigned int vr_op)
...
+    first_rate_iter = true;
+    last_timer_value = GetTickCount64();
+
+    while (true) {
+        Sleep(1000);
+
+        /* Size of the terminal does not limit us on Windows. */
+        print_number_interface = MAX_OUTPUT_IF;
+
+        ret = system("cls");
```
Comment (Attil) [inline utils/vif.c]: Would clearing the screen in another way be more efficient? calling `system("cls");` causes quite a bit flickering.

I mean something like this: `https://support.microsoft.com/en-us/help/99261/how-to-performing-clear-screen-cls-in-a-console-application`

**Pull request merged**